hummingbot/connector/exchange/bybit/bybit_api_order_book_data_source.py

```diff
@@ -0,0 +1,315 @@
+from collections import defaultdict
+
+import aiohttp
+import asyncio
+import logging
+
+from typing import (
+    Dict,
+    List,
+    Optional, Any,
+)
+
+from hummingbot.connector.exchange.bybit import bybit_constants as CONSTANTS, bybit_utils
+from hummingbot.core.data_type.order_book import OrderBook, OrderBookMessage
+
+from hummingbot.connector.exchange.bybit.bybit_order_book import BybitOrderBook
+from hummingbot.connector.exchange.bybit.bybit_websocket_adaptor import BybitWebSocketAdaptor
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.logger import HummingbotLogger
+
+
+class BybitAPIOrderBookDataSource(OrderBookTrackerDataSource):
+    _ORDER_BOOK_SNAPSHOT_DELAY = 60 * 60
+
+    _logger: Optional[HummingbotLogger] = None
+    _trading_pair_symbol_map: Dict[str, Dict[str, str]] = {}
```

Comment:
After discussing this with Daniel we arrived to the conclusion that it might be a good idea to have the map instead of the function that assumes a conversion logic to avoid associating the client logic with any current scenario. If the exchanges adds more currencies in the future with codes longer than 4 characters, the hardcoded logic would fail, but the map built based on the symbols endpoint information will keep working correctly.